### PR TITLE
Add support for Kafka 3.3.1 and remove Kafka 3.1.x

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Task

### Description

This PR adds support for Kafka 3.3.1 and removes support for 3.1.x versions.

_This PR currently uses the RC0 of Kafka 3.3.1._

### Checklist

- [x] Make sure all tests pass
- [x] Update documentation
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

File: kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java [hunk 1]
```diff
@@ -24,9 +25,13 @@
  * The presence of this file is tested via a Kube "exec" readiness probe to determine when the broker is ready.
  */
 public class KafkaAgent {
-
     private static final Logger LOGGER = LoggerFactory.getLogger(KafkaAgent.class);
 
+    // KafkaYammerMetrics class in Kafka 3.3+
+    private static final String YAMMER_METRICS_IN_KAFKA_3_3_AND_LATER = "org.apache.kafka.server.metrics.KafkaYammerMetrics";
+    // KafkaYammerMetrics class in Kafka 3.2-
+    private static final String YAMMER_METRICS_IN_KAFKA_3_2_AND_EARLIER = "kafka.metrics.KafkaYammerMetrics";
```
Comment (ppatierno) [inline kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java:33]: just curious, is that normal it doesn't have a domain prefix like `org.apache`?

**Review by ppatierno — APPROVED** [review]
LGTM. Just a nit/curiosity.

File: kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java (hunk 1 above)
Comment (scholzj) [reply kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java:33]: That is how the whole Kafka `core` code written in Scala uses as a package. So not much we can do about it.

**Pull request merged**